Update requirements.mdx

mrpavan opened the pull request:
Provider source address delimiter separator / was within the hostname portion before closing bracket ']' and same has been corrected
file - /main/website/docs/language/providers/requirements.mdx

Original Line 120
`[<HOSTNAME>/]<NAMESPACE>/<TYPE>`

Updated
`[<HOSTNAME>]/<NAMESPACE>/<TYPE>`

<!--

Describe in detail the changes you are proposing, and the rationale.

See the contributing guide:

https://github.com/hashicorp/terraform/blob/main/.github/CONTRIBUTING.md

-->

<!--

Link all GitHub issues fixed by this PR, and add references to prior
related PRs.

-->

Fixes #33911

## Target Release

<!--

In normal circumstances we only target changes at the upcoming minor
release, or as a patch to the current minor version. If you need to
port a security fix to an older release, highlight this here by listing
all targeted releases.

If targeting the next patch release, also add the relevant x.y-backport
label to enable the backport bot.

-->

1.5.x

## Draft CHANGELOG entry

<!--

Choose a category, delete the others:

-->

### UPGRADE NOTES 

<!--

Write a short description of the user-facing change. Examples:

- `terraform show -json`: Fixed crash with sensitive set values.
- When rendering a diff, Terraform now quotes the name of any object attribute whose string representation is not a valid identifier.
- The local token configuration in the cloud and remote backend now has higher priority than a token specified in a credentials block in the CLI configuration.

--> 

-  Provider source address delimiter placement correction from `[<HOSTNAME>/]<NAMESPACE>/<TYPE>` to `[<HOSTNAME>]/<NAMESPACE>/<TYPE>`

hashicorp-cla commented:
[![CLA assistant check](https://cla.hashicorp.com/pull/badge/not_signed)](https://cla.hashicorp.com/hashicorp/terraform?pullRequest=33912)


Thank you for your submission! We require that all contributors sign our Contributor License Agreement ("CLA") before we can accept the contribution. [Read and sign the agreement](https://cla.hashicorp.com/hashicorp/terraform?pullRequest=33912)

[Learn more about why HashiCorp requires a CLA and what the CLA includes](https://www.hashicorp.com/cla)

<sub>Have you signed the CLA already but the status is still pending? [Recheck it](https://cla.hashicorp.com/check/hashicorp/terraform?pullRequest=33912).</sub>

alisdair reviewed (changes requested):
Thanks for the contribution, but this is not correct. Please see the discussion in the original issue. While the hostname is optional, if omitted, the leading slash must not be present.

The clearest way I can imagine to explain the source address format was mentioned in the linked issue: give two examples, one with the hostname, and one without. This will require some rewording of this section of the docs.

mrpavan commented:
As recommended by the @alisdair @apparentlymart , reverted provider format and added example for valid provider formats as below 

Examples of valid provider formats include 
- `NAMESPACE/TYPE`
- `HOSTNAME/NAMESPACE/TYPE` 

Thanks,
Pavan

alisdair commented on website/docs/language/providers/requirements.mdx:
```diff
@@ -117,7 +117,11 @@ primary location where Terraform can download it.
 follows:
 
-`[<HOSTNAME>/]<NAMESPACE>/<TYPE>`
+`[<HOSTNAME/>]<NAMESPACE>/<TYPE>`
```
I think this was a typo in your revert:

```suggestion
`[<HOSTNAME>/]<NAMESPACE>/<TYPE>`
```

alisdair commented on website/docs/language/providers/requirements.mdx:
```diff
@@ -117,7 +117,11 @@ primary location where Terraform can download it.
-`[<HOSTNAME>/]<NAMESPACE>/<TYPE>`
+`[<HOSTNAME/>]<NAMESPACE>/<TYPE>`
+
+Examples of valid provider formats include 
```
```suggestion
Examples of valid provider source address formats include:
```

alisdair commented on website/docs/language/providers/requirements.mdx:
```diff
@@ -117,7 +117,11 @@ primary location where Terraform can download it.
+
+Examples of valid provider formats include 
+- `NAMESPACE/TYPE`
+- `HOSTNAME/NAMESPACE/TYPE` 
```
```suggestion
- `HOSTNAME/NAMESPACE/TYPE`
```

alisdair reviewed (commented):
Thanks! I have a couple of formatting suggestions inline, but after that this looks good to merge to me.
